libraries/chain/include/eosio/chain/wasm_interface_private.hpp

```diff
@@ -38,12 +38,14 @@ namespace eosio { namespace chain {
       struct by_last_block_num;
 
       wasm_interface_impl(wasm_interface::vm_type vm, const chainbase::database& d) : db(d) {
+#ifdef EOSIO_WAVM_RUNTIME_ENABLED
          if(vm == wasm_interface::vm_type::wavm)
             runtime_interface = std::make_unique<webassembly::wavm::wavm_runtime>();
-         else if(vm == wasm_interface::vm_type::wabt)
+#endif
+         if(vm == wasm_interface::vm_type::wabt)
             runtime_interface = std::make_unique<webassembly::wabt_runtime::wabt_runtime>();
-         else
-            EOS_THROW(wasm_exception, "wasm_interface_impl fall through");
+         if(!runtime_interface)
+            EOS_THROW(wasm_exception, "${r} wasm runtime not supported on this platform and/or configuration", ("r", vm));
```

Comment:
Do we want to distinguish between unsupported runtimes and totally non-existent runtimes in the error message?

Reply:
Unless I'm misunderstanding something, since this function works on an enum, I think the only way you get here is for an unsupported runtime. If you `--wasm banana` you'll fail much earlier when the string fails to be converted to an enum.